WikiPreview.Fluent.Plugin/WikiPreviewSearchApp.cs

```diff
@@ -108,59 +109,88 @@ public ValueTask LoadSearchApplicationAsync()
-                        string pageId = value.PageId.ToString();
-                        string wikiUrl = WikiRootUrl + displayedName;
-                        BitmapImageResult bitmapImageResult;
-
-                        string additionalInfo = "";
-                        if (!string.IsNullOrWhiteSpace(resultName))
                         {
-                            using var reader = new StringReader(resultName);
-                            additionalInfo = await reader.ReadLineAsync().ConfigureAwait(false) ?? resultName;
-                        }
-
-                        if (value.Thumbnail != null)
-                        {
-                            string imgUrl = value.Thumbnail.Source;
-                            using var imageClient = new HttpClient();
-                            Stream stream = await imageClient.GetStreamAsync(imgUrl, cancellationToken)
+                            WikiPreviewSearchResult wikiPreviewSearchResult =
+                                await GenerateSearchResult(entry.Value, searchedText).ConfigureAwait(false);
+                            await channel.Writer.WriteAsync(wikiPreviewSearchResult, CancellationToken.None)
                                 .ConfigureAwait(false);
-                            bitmapImageResult = new BitmapImageResult(new Bitmap(stream));
-                        }
-                        else
-                        {
-                            bitmapImageResult = new BitmapImageResult(); // create empty if no image source.
-                        }
-
-                        WikiPreviewSearchResult wikiPreviewSearchResult = new()
-                        {
-                            Url = wikiUrl,
-                            PreviewImage = bitmapImageResult,
-                            DisplayedName = displayedName,
-                            ResultName = resultName,
-                            SearchedText = searchedText,
-                            Score = score,
-                            PageId = pageId,
-                            AdditionalInformation = additionalInfo
-                        };
-                        await channel.Writer.WriteAsync(wikiPreviewSearchResult).ConfigureAwait(false);
-                    }, cancellationToken).ContinueWith(_ => channel.Writer.Complete());
+                        }, cancellationToken)
+                        .ContinueWith(_ => channel.Writer.Complete(), CancellationToken.None);
                 else
                     channel.Writer.Complete();
-            });
+            }, CancellationToken.None);
 
             await foreach (WikiPreviewSearchResult item in channel.Reader.ReadAllAsync(cancellationToken))
                 yield return item;
         }
 
-        public ValueTask<ISearchResult> GetSearchResultForId(string serializedSearchObjectId)
+        public ValueTask<WikiPreviewSearchResult> GetSearchResultForId(string serializedSearchObjectId)
```

Comment:
You are not implementing the right method, the right signature of the `interface` is with `object` and not `string`. 
It compiles because in `ISearchApplication` the method has `default` implementation (meaning you don't have to implement it.

Reply:
Thank you. I've changed it to `object`. And finally returning the `WikiPreviewSearchResult` after await.

Reply:
This is still not working, kindly let me know on this further.